kedro-datasets/kedro_datasets/polars/generic_dataset.py

```diff
@@ -0,0 +1,258 @@
+"""``GenericDataSet`` loads/saves data from/to a data file using an underlying
+filesystem (e.g.: local, S3, GCS). It uses polars to handle the
+type of read/write target.
+"""
+from copy import deepcopy
+from io import BytesIO
+from pathlib import PurePosixPath
+from typing import Any, Dict
+
+import fsspec
+import polars as pl
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    DataSetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+ACCEPTED_WRITE_MODES = ["overwrite", "ignore"]
+
+ACCEPTED_WRITE_FILE_FORMATS = [
+    "csv",
+    "ipc",
+    "parquet",
+    "json",
+    "ndjson",
+    "avro",
+]
+# always a superset of ACCEPTED_WRITE_FILE_FORMATS
+ACCEPTED_READ_FILE_FORMATS = ACCEPTED_WRITE_FILE_FORMATS + [
+    "excel",
+    "delta",
+]
+
+
+# pylint: disable=too-many-instance-attributes
+class GenericDataSet(AbstractVersionedDataSet[pl.DataFrame, pl.DataFrame]):
+    """`polars.GenericDataSet` loads/saves data from/to a data file using an underlying
+    filesystem (e.g.: local, S3, GCS). It uses polars to dynamically select the
+    appropriate type of read/write target on a best effort basis.
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+    .. code-block:: yaml
+        cars:
+          type: polars.GenericDataSet
+          file_format: csv
```

Comment:
can we do and example of something not possible with your other dataset e.g. parquet?

Reply:
Sure.